File: game-of-life-cpp/Wolf.cpp
#include "include/Wolf.h"

Wolf::Wolf(World* world, Point& position)
    : Animal(world, 9, 5, position, "🐺") {
}

Wolf::Wolf(World* world, Point& position, int age)
    : Animal(world, 9, 5, position, "🐺", age) {
}

Wolf::~Wolf() {

}

File: game-of-life-java/src/Wolf.java
import javax.swing.*;
import java.awt.*;

public class Wolf extends Animal {
    private static final ImageIcon wolfIcon = new ImageIcon("images/wolf.png");
    public static final Image scaledWolfIcon = wolfIcon.getImage().getScaledInstance(30, 30, Image.SCALE_SMOOTH);

    public Wolf(Point position, World world, int age) {
        super(9, 5, position, world, age);
    }

    @Override
    public ImageIcon getIcon() {
        return wolfIcon;
    }
}
